src/core/render/index.js

```diff
@@ -41,58 +43,179 @@ function formatUpdated(html, updated, fn) {
+      // <button v-on:click="doThis"></button>
+      // <button v-on:click.once="doThis"></button>
+      // <button v-on:[event]="doThis"></button>
+      // <button @click.stop.prevent="doThis">
+      // <a :href="url">
+      // <a :[key]="url">
+      const reHasDirective = /<[^>/]+\s([@:]|v-)[\w-:.[\]]+[=>\s]/;
+
+      vueMountData.push(
+        ...dom
+          .findAll('.markdown-section > *')
+          // Remove duplicates
+          .filter(elm => !vueMountData.some(([e, c]) => e === elm))
+          // Detect Vue content
+          .filter(elm => {
+            const isVueMount =
+              // is a component
+              elm.tagName.toLowerCase() in
+                (docsifyConfig.vueComponents || {}) ||
+              // has a component(s)
+              elm.querySelector(vueComponentNames.join(',') || null) ||
+              // has curly braces
+              reHasBraces.test(elm.outerHTML) ||
+              // has content directive
+              reHasDirective.test(elm.outerHTML);
+
+            return isVueMount;
+          })
+          .map(elm => {
+            // Clone global configuration
+            const vueConfig = Object.assign(
+              {},
+              docsifyConfig.vueGlobalOptions || {}
+            );
+
+            // Replace vueGlobalOptions data() return value with shared data object.
+            // This provides a global store for all Vue instances that receive
+            // vueGlobalOptions as their configuration.
+            if (vueGlobalData) {
+              vueConfig.data = function() {
+                return vueGlobalData;
+              };
+            }
+
+            return [elm, vueConfig];
+          })
+      );
+    }
 
-          app.mount(elm);
-        }
-      }
-      // Vue 2
-      else if (vueVersion === 2) {
-        const isAlreadyVue = Boolean(elm.__vue__ && elm.__vue__._isVue);
-
-        if (!isAlreadyVue) {
-          new window.Vue({
-            mounted: function() {
-              this.$destroy();
-            },
-          }).$mount(elm);
+    // Mount
+    for (const [mountElm, vueConfig] of vueMountData) {
+      const isVueAttr = 'data-isvue';
+      const isSkipElm =
+        // Is an invalid tag
+        mountElm.matches('pre, script') ||
+        // Is a mounted instance
+        isMountedVue(mountElm) ||
+        // Has mounted instance(s)
+        mountElm.querySelector(`[${isVueAttr}]`);
+
+      if (!isSkipElm) {
+        mountElm.setAttribute(isVueAttr, '');
+
+        if (vueVersion === 2) {
+          vueConfig.el = undefined;
+          new window.Vue(vueConfig).$mount(mountElm);
+        } else if (vueVersion === 3) {
+          const app = window.Vue.createApp(vueConfig);
+
+          // Register global vueComponents
+          vueComponentNames.forEach(name => {
+            const config = docsifyConfig.vueComponents[name];
+
+            app.component(name, config);
+          });
+
+          app.mount(mountElm);
```

Comment:
I haven't tried Vue 3, not sure if it has the same issue as the comment about the blue border with Vue 2.

Reply:
No issue. 😄